### PR TITLE
Add more parameters (androidAppId and iosBundleId) in devtools events. 

### PR DESCRIPTION
**Pull request by hangyujin**
Devtool PR: https://github.com/flutter/devtools/pull/8177
 
---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/lib/src/event.dart
Comment (andrewkolos) [inline pkgs/unified_analytics/lib/src/event.dart:402]:
```suggestion
    // DeepLinkScreenMetrics
```

**Review by andrewkolos — APPROVED** [review]
RSLGTM

**Comment by kenzieschmoll** [comment]
I am hesitant to add the `iosBundleId` parameter until we have more clarity around the PDD update approval, because then removing this parameter later would be a breaking change to `unified_analytics`.

**Comment by hangyujin** [comment]
PDD updates for deep link validation and impact tracking is approved, internal tracker: b/362324447.

reopening this PR

File: pkgs/unified_analytics/CHANGELOG.md
Comment (kenzieschmoll) [inline pkgs/unified_analytics/CHANGELOG.md:4]: super nit: use back tics and code references

 - Added `androidAppId` and `iosBundleId` to the `Event.devtoolsEvent` constructor.

**Review by kenzieschmoll — APPROVED** [review]

File: pkgs/unified_analytics/CHANGELOG.md
Comment (kenzieschmoll) [inline pkgs/unified_analytics/CHANGELOG.md:1]: if you want this published, you'll need to remove the `-wip` from the version here and in the pubspec.

**Comment by hangyujin** [comment]
This looks like it stucks at "3 workflows awaiting approval"

**Comment by hangyujin** [comment]
The checks have passed! 

I don't have write access to  merge this PR.

**Pull request merged**
